flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -917,8 +917,15 @@ class JobManager(
     case ScheduleOrUpdateConsumers(jobId, partitionId) =>
       currentJobs.get(jobId) match {
         case Some((executionGraph, _)) =>
-          sender ! decorateMessage(Acknowledge)
-          executionGraph.scheduleOrUpdateConsumers(partitionId)
+          try {
+            executionGraph.scheduleOrUpdateConsumers(partitionId)
+            sender ! decorateMessage(Acknowledge)
+          } catch {
+            case e: ExecutionGraphException =>
```

Comment:
Does it make sense to catch the more generic `Exception` type here in order to make the sender notice any problems sooner? I see that the method only throws EGExceptions currently but maybe at some point in time someone introduces a runtime exception etc. This would only be logged at the JM and the task's ask would timeout.